ui/application-window.ui

```diff
@@ -50,6 +95,126 @@ Author: Philipp Wolfer <ph.wolfer@gmail.com>
     <property name="show_menubar">False</property>
     <signal name="draw" handler="on_window_draw" swapped="no"/>
     <signal name="screen-changed" handler="on_window_screen_changed" swapped="no"/>
+    <child type="titlebar">
+      <object class="GtkHeaderBar" id="headerbar">
+        <property name="visible">True</property>
+        <property name="can_focus">False</property>
+        <property name="has_subtitle">False</property>
+        <property name="spacing">0</property>
```

Comment:
You'll want to set the spacing to 6px, otherwise the separator at the end, the menu button, and the close button will all have no padding between them.

```suggestion
        <property name="spacing">6</property>
```